documentation/development-docs/enable-subscription-id-overriding.md

```diff
@@ -1,12 +1,104 @@
+  - [Regenerate Help Documents](#regenerate-help-documents)
+- [Notes / Troubleshooting](#notes--troubleshooting)
+  - [Your Cmdlet Implements `IDynamicParameters`](#your-cmdlet-implements-idynamicparameters)
+  - [Static Analysis Fails: Parameters are Removed](#static-analysis-fails-parameters-are-removed)
 
 ## Background
 
+Working with **multiple Azure subscriptions** can be inconvenient in Azure PowerShell, because users has to keep switching the context, which is why we introduced a mechanism that can easily turn your module **multiple-subscription-friendly** in [Az v6.4.0](https://github.com/Azure/azure-powershell/blob/isra-fel-patch-1/ChangeLog.md#640---september-2021). For example
+
+```powershell
+# Legacy (switch subscription first)
+Select-AzSubscription -SubscriptionId "00000000-0000-0000-0000-000000000000"
+New-AzAksCluster ...
+
+# New (just -SubscriptionId)
+New-AzAksCluster ... -SubscriptionId "00000000-0000-0000-0000-000000000000"
+```
+
+The new design does not only simplify scripts, but also runs more efficiently, as it saves 1 cmdlet execution per subscription switching.
+
 ## Steps
 
-1. {add attribute}
-2. if any cmdlet implememts IDynamicParameter
+Here are the two simple steps to enable this for your module:
+
+### Add `SupportsSubscriptionId` Attribute
+
+You can either add it to the cmdlet base class, which applies to all the derived cmdlets; or to individual cmdlets.
+
+```csharp
+[SupportsSubscriptionId] // Adding to base class: all cmdlets that inherit `KubeCmdletBase` will benefit
+public abstract class KubeCmdletBase : AzureRMCmdlet { /* ... */ }
+
+[SupportsSubscriptionId] // Adding to cmdlet class: only this cmdlet will benefit
+public class NewAzureRmAks : CreateOrUpdateKubeBase { /* ... */ }
+```
+
+### Regenerate Help Documents
+
+By adding the attribute your cmdlet(s) get a `[-Subscription <String>]` parameter, so the help documents need to be regenerated.
+Please refer to [Azure PowerShell Help Generation](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md) for more details.
+
+
+## Notes / Troubleshooting
+
+This design is compatible with most modules, however, there are some uncommon case you should be careful.
+
+### Your Cmdlet Implements `IDynamicParameters`
+
+The `-SubscriptionId` parameter is added via `IDynamicParameters` interface, so if your cmdlet has already implemented it, make sure:
+
+1. `GetDynamicParameters()` is decorated with [`new` modifier](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/new-modifier).
+1. Call `base.GetDynamicParameters()` and combine the results in your `GetDynamicParameters()`.
+
+A sample implementation:
+
+```csharp
+public new object GetDynamicParameters()
+{
+    var parameters = base.GetDynamicParameters() as RuntimeDefinedParameterDictionary;
+    AzureRmProfile localProfile = DefaultProfile as AzureRmProfile;
```

Comment:
Do these mean a cmdlet should add Line64 and then implement its own features like Line 65-76 if it has implemented `IDynamicParameters`

Reply:
Yes you are right, I should probably make 65-76 more abstract so it won't attract too much attention

Reply:
Hi @BethanyZhou I rewrote the code sample. Please review.  Thanks!